constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/Transition.java

```diff
@@ -710,11 +710,31 @@ public void addCustomColor(int state, String widgetId, String property, int colo
     public void updateFrom(ConstraintWidgetContainer container, int state) {
         final ArrayList<ConstraintWidget> children = container.getChildren();
         final int count = children.size();
+        WidgetState [] states = new WidgetState[count];
+        boolean relative = false;
         for (int i = 0; i < count; i++) {
             ConstraintWidget child = children.get(i);
             WidgetState widgetState = getWidgetState(child.stringId, null, state);
+            states[i] = widgetState;
             widgetState.update(child, state);
+           String id =  widgetState.getPathRelativeId();
+            if (id != null) {
+                relative = true;
```

Comment:
Seems like we could just have `widgetState.setPathRelative(getWidgetState(id, null, state));` here instead.
No need to loop twice.